dashboard/test/mailers/ops_mailer_test.rb

```diff
@@ -15,4 +15,114 @@ class OpsMailerTest < ActionMailer::TestCase
     assert_match "added 1 teachers", mail.body.encoded
   end
 
+  test "workshop_in_2_weeks_reminder" do
+    # Create a workshop that starts 14 days from now. Include other segments for authenticity
+    @workshop_1 = create(:workshop, phase: 2, cohorts: [create(:cohort, id: 1,
+      teachers: [create(:teacher, email: 'abc@code.org'), create(:teacher, email: 'xyz@code.org')])],
+      facilitators: [create(:facilitator, email: 'facilitator_1@code.org')])
+    @workshop_1.segments.clear
+    @workshop_1.segments << create(:segment, workshop_id: @workshop_1.id, start: DateTime.now + 14.day, end: DateTime.now + 14.day + 60.minute)
+    @workshop_1.segments << create(:segment, workshop_id: @workshop_1.id, start: DateTime.now + 15.day, end: DateTime.now + 15.day + 60.minute)
```

Comment:
Here's a bug: The Workshop class isn't using the ruby time but rather CURDATE() from the database.  This is an issue because Ruby and the database might disagree on what date it is.  If you got unlucky on your cron job scheduling, this could even lead to duplicated and missing reminder emails.  Suppose the server was busy and ran at 12:03am one day and at 12:00am the next day.  Suppose the database server is two minutes behind the cron server.  Then in database time the cron job would run at 12:01am and again at 11:57pm on the *same* day, resulting in doubled emails to all of the participants.  And no one would get emails the following day.

Could you add a TODO to workshop.rb. to fix this bug, and verify that the cron job doesn't run any time close to midnight?